win32ss/user/ntuser/ime.c

```diff
@@ -40,6 +41,12 @@ typedef struct tagIMEHOTKEY
 PIMEHOTKEY gpImeHotKeyList = NULL;
 LCID glcidSystem = 0;
 
+static inline PIMEUI FASTCALL IntGetImeUIFromWnd(_In_ PWND pwnd)
+{
+    // (pwnd + 1) points to the window extra bytes where the PIMEUI pointer is stored.
+    return (PIMEUI)ReadUnalignedUlongPtr((const ULONG_PTR *)(pwnd + 1));
```

Comment:
```suggestion
    ASSERT(pwnd->cbwndExtra >= sizeof(PIMEUI));
    return (PIMEUI)ReadUnalignedUlongPtr((const ULONG_PTR *)(pwnd + 1));
```
or with a `==` in the test. Also, either ASSERT (if you are 100% sure the window is going to only store a PIMEUI pointer), otherwise something like:
```
    if (pwnd->cbwndExtra < sizeof(PIMEUI))
        return NULL;
```